src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncedTxGossipPolicy.cs

```diff
@@ -0,0 +1,19 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.TxPool;
+
+namespace Nethermind.Synchronization.ParallelSync;
+
+public class SyncedTxGossipPolicy : ITxGossipPolicy
+{
+    private readonly ISyncModeSelector _syncModeSelector;
+
+    public SyncedTxGossipPolicy(ISyncModeSelector syncModeSelector)
+    {
+        _syncModeSelector = syncModeSelector;
+    }
+
+    public bool CanGossipTransactions => (_syncModeSelector.Current & SyncMode.WaitingForBlock) != 0;
```

Comment:
Will this also cover during block production/validation? (as the pool will be dumped after it)

Reply:
Currently not. Not sure if it is a good idea to stop those messages completely during this phase - probably it could be more sophisticated like storing hashes we get in a temporary collection and then trying to get them when resuming.

Reply:
We could cache the value based on `SyncModeChanged` event, but this is just a flag check so not sure if worth it.